rs/xnet/payload_builder/src/certified_slice_pool.rs

```diff
@@ -377,7 +377,19 @@ impl Payload {
         message_limit: Option<usize>,
         byte_limit: Option<usize>,
     ) -> CertifiedSliceResult<(Option<Self>, Option<Self>)> {
-        let message_limit = message_limit.unwrap_or(usize::MAX);
+        // `messages_end` may not exceed a certain `max_messages_end` in order to cap the number of
+        // signals in the reverse stream. Calculate the maximum number of messages we can include
+        // in the slice such that this is respected.
+        let max_message_limit = {
+            let messages_begin =
+                self.messages_begin().unwrap_or(self.header.begin()).get() as usize;
+            let max_messages_end = max_messages_end(self.header.begin()).get() as usize;
```

Comment:
Personal opinion: I'd consider naming the function and variable something like `max_message_index`. "Max messages end" requires an extra level of indirection: "messages end" is the end of the slice, we don't want the slice to go beyond some index. But basically, all we want to say is "don't induct anything past this message index". (That this results in a limit on a slice's `messages_end` is a side-effect.)

Reply:
`max_messages_index()` it is.

Reply:
Done.